### PR TITLE
TST: Simplify and clarify test skipping

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

File: pandas/tests/plotting/test_backend.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def test_setting_backend_without_plot_raises(monkeypatch):
     assert pandas.options.plotting.backend == "matplotlib"
 
 
-@td.skip_if_mpl
+@td.skip_if_installed("matplotlib")
```
Comment (jbrockmendel) [inline pandas/tests/plotting/test_backend.py:83]: is there any perf impact of evaluating these once vs in a bunch of places?

**Comment by jbrockmendel** [comment]
One question, otherwise LGTM

File: pandas/tests/arithmetic/test_datetime64.py [hunk 2]
```diff
@@ -414,7 +414,7 @@ def test_dti_cmp_datetimelike(self, other, tz_naive_fixture):
         dti = date_range("2016-01-01", periods=2, tz=tz)
         if tz is not None:
             if isinstance(other, np.datetime64):
-                pytest.skip("no tzaware version available")
+                pytest.skip(f"{type(other.__name__)} is not tz aware")
```
Comment (mroeschke) [inline pandas/tests/arithmetic/test_datetime64.py:417]:
```suggestion
                pytest.skip(f"{type(other).__name__} is not tz aware")
```

File: pandas/tests/plotting/test_backend.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/plotting/test_backend.py:83]: Very slight. e.g. when running `test_eval.py` with `skip_if_no_ne` replaced locally I'm getting a runtime of 26.86s on this PR vs 26.69s on main. IMO I think this timing difference is OK given the cleanup

**Pull request merged**